slackClient.js

```diff
@@ -10,48 +7,28 @@ class SlackClient {
   /**
    * @param {Object} params
    */
-  constructor(params) {
-    this.events = params.slackEvents;
-    this.slack = params.slackWebClient;
-
-    this._setupListeners();
+  constructor(slackEventsAdapter, slackWebClient) {
+    this.events = slackEventsAdapter;
+    this.slack = slackWebClient;
   }
 
   /**
-   * @private
+   * Adds a listener for a slack event
+   *
+   * @param {string} eventType
+   * @param {function} callback
    */
-  _setupListeners() {
-    this.events.on('app_mention', (message, body) => {
-      // Only respond to messages that have no subtype (plain messages)
-      if (!message.subtype) {
-        const search = this._extractSearchText(message.text);
-
-        if (!search) {
-          this._postMessage(message.channel, 'Invalid search: try "@Sakura <Insert Anime Name>"');
-        } else {
-          this._postMessage(message.channel, `Searching for anime "${search}"...`);
-          this._searchAnime(message.channel, search);
-        }
-      }
-    });
-
-    this.events.on('error', (error) => {
-      if (error.code === slackEventsApi.errorCodes.TOKEN_VERIFICATION_FAILURE) {
-        console.error(`An unverified request was sent to the Slack events Request URL. Request body: ${JSON.stringify(error.body)}`);
-      } else {
-        console.error(`An error occurred while handling a Slack event: ${error.message}`);
-      }
-    });
+  onEvent(eventType, callback) {
+    this.events.on(eventType, callback);
```

Comment:
In the future, it'd be nice if we can manage the events instead of only binding them to events. If we could unbind them that'd be cool too